test/kubeclient/kubeclient.go

```diff
@@ -0,0 +1,268 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeclient
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/Mirantis/hmc/test/utils"
+	. "github.com/onsi/ginkgo/v2"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	awsCredentialsSecretName = "aws-variables"
+)
+
+type KubeClient struct {
+	Namespace string
+
+	Client         kubernetes.Interface
+	ExtendedClient apiextensionsclientset.Interface
+	Config         *rest.Config
+}
+
+// NewFromLocal creates a new instance of KubeClient from a given namespace
+// using the locally found kubeconfig.
+func NewFromLocal(namespace string) (*KubeClient, error) {
```

Comment:
Isn't it easier to use [controller runtime client](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client)?

Reply:
Matter of opinion I guess, I felt for writing tests client-go's lack of abstraction made it easier.  I've written operators and worked with both clients and find it easier to work with client-go.

Reply:
Fair enough. I just have a feeling certain stuff (like getting config) already exists in controller runtime client. 
But I guess it's really a matter of taste here.
@Kshatrix WDYT?